File: factory/LocalPaymentMethodFactory.java
package factory;

public class LocalPaymentMethodFactory implements PaymentMethodFactory {
    @Override
    public PaymentMethod createPaymentMethod(CardType cardType, String cardHolder, String cardNumber,
                                             String cvv, String expiryDate) {
        if (cardType == CardType.DINERS) {
            return new Diners(cardHolder, cardNumber, cvv, expiryDate);
        }
        if (cardType == CardType.CARTE_BANCAIRE) {
            return new CarteBancaire(cardHolder, cardNumber, cvv, expiryDate);
        }
        throw new IllegalArgumentException("Card type not supported...");
    }
}

File: typescript/patterns/factory/local-payment-method-factory.ts
import { CardType } from "./card-type";
import { CarteBancaire } from "./carte-bancaire";
import { Diners } from "./Diners";
import { PaymentMethod, PaymentMethodFactory } from "./payment-method-factory";

export class LocalPaymentMethodFactory implements PaymentMethodFactory {
    createPaymentMethod(
      cardType: CardType,
      cardHolder: string,
      cardNumber: string,
      cvv: string,
      expiryDate: string
    ): PaymentMethod {
      if (cardType === CardType.DINERS) {
        return new Diners(cardHolder, cardNumber, cvv, expiryDate);
      }
      if (cardType === CardType.CARTE_BANCAIRE) {
        return new CarteBancaire(cardHolder, cardNumber, cvv, expiryDate);
      }
      throw new Error("Card type not supported...");
    }
  }